rpc/handlers.go

```diff
@@ -1379,6 +1395,82 @@ func (h *Handler) SpecVersion() (string, *jsonrpc.Error) {
 	return "v0.5.0", nil
 }
 
+func randomID() uint64 {
+	var n uint64
+	for err := binary.Read(rand.Reader, binary.LittleEndian, &n); err != nil; {
+	}
+	return n
+}
+
+func (h *Handler) SubscribeNewHeads(ctx context.Context) (uint64, *jsonrpc.Error) {
+	w, ok := jsonrpc.ConnFromContext(ctx)
+	if !ok {
+		return 0, jsonrpc.Err(jsonrpc.MethodNotFound, nil)
+	}
+
+	id := randomID()
+	subscriptionCtx, subscriptionCtxCancel := context.WithCancel(ctx)
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	h.subscriptions[id] = &subscription{
+		cancel: subscriptionCtxCancel,
+		conn:   w,
+	}
+
+	go func() {
+		sink := make(chan *core.Header)
```

Comment:
Can you take a look at how `event.Subscription.Send()` behaves when one of the sinks are full/non-buffered (ie blocks on write)

Reply:
It sends on each channel that is subscribed, blocking if it tries to send on a channel that is full.

Reply:
So, our `Store()` might block for arbitrarily long if some of our WS connections are busy?

Reply:
Yes. A small buffer sounds reasonable. What do you think about `3`?